### PR TITLE
Add contribution guidelines

### PR DESCRIPTION
**Pull request by madchutney**
* New contribution markdown file.
* Add licence and contribution sections to readme and link to appropriate files.
* Trailing whitespace removed.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Contributing
+
+Mbed Cloud SDK for JavaScript is an open source project of Arm Mbed.
```
Comment (thegecko) [inline CONTRIBUTING.md]: `from` Arm Mbed?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## How to Contribute Code
+
+Please keep contributions small and independent. We would much rather have
+multiple pull requests for each cool thing you've done rather than have them all
+in the same one. This will help us review, give feedback and merge in your
+changes.
+
+- Fork the repository.
+- Make your change and write unit tests, please try to match the existing coding
```
Comment (thegecko) [inline CONTRIBUTING.md]: 'please __do__ match existing code styling' :)

File: README.md [hunk 3]
```diff
@@ -18,12 +18,23 @@ $ npm install mbed-cloud-sdk
 
 `/node_modules/mbed-cloud-sdk` now contains:
 
-* `bundles` - minified browser scripts. 
-* `lib` - Node.js modules.
```
Comment (thegecko) [inline README.md]: thanks

File: README.md [hunk 4]
```diff
@@ -18,12 +18,23 @@ $ npm install mbed-cloud-sdk
...
 
 Please refer to the examples folder for some node and web examples.
+
+## Contributing
+
+Mbed Cloud SDK for JavaScript is open source and we would like your help; there is a
+brief guide on how to get started in [CONTRIBUTING.md](CONTRIBUTING.md).
+
+## Licence
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
```
Comment (thegecko) [inline README.md]: `license` or `licence`? We have both here :)

Can we put the license badge at the top and perhaps link to the SPDX license?

Comment (madchutney) [reply README.md]: I guess we better stick with the US spelling since it is the name of the file.

**Comment by madchutney** [comment]
I wasn't sure about mentioning that the test must pass without providing some guidance on how to run them.

**Pull request merged**